### ISSUE TITLE
Automating conan package build process based on source code change of library.

### ISSUE THREAD

**Issue by himansu2906** [conan-io/conan#11976]
Is there any specific flow present to automate whole flow of creation of conan package?

**Scenario**: while creating a project build, if git recognize changes in the the source code of library, it should automatically create new version of package and upload it to conan server as well for future use.

**Comment by memsharded** [comment]
Hi @himansu2906 

This is a typical CI flow, not a Conan one. There are 2 possible scenarios:

- If the recipe conanfile.py is in the source repo (to package your own libraries), then your own CI will detect whatever event you decide (like a new release tag), and use it to do the ``conan create . <version>@`` in your CI
- If your conanfile.py is in another repo, you need some kind of subscribing mechanism to the repo that contains the source code, to be notified about new releases. Then, your CI can be fired and do the job. This will typically require a bit more of automation, as the source url, commit, etc is in ``conandata.yml`` file, and it will have to be updated by the CI to point to the new release data.

To summarize: this is outside of the scope of a package manager, it is typically a CI responsibility

**Comment by memsharded** [comment]
I am closing this question as responded, but please don't hesitate to open new issues for any further follow up or question you might have. Thanks!
